Remove redundant migration screen.

markbiek opened the pull request:
<!--
Link a related issue to this PR. If the PR does not immediately resolve the issue,
for example, it requires a separate deployment to production, avoid
using the "fixes" keyword and instead attach the [Status] Fix Inbound label to
the linked issue.
-->

Related to #101017

## Proposed Changes

This updates the site-migration-flow so that we go directly from the "authorize application" or "share credentials screen" to `/overview/:siteSlug`. Previously, we went to the `importer-migrate-message` step which showed essentially the same message as `/overview`.

Note that it's necessary to reset site data when redirecting to the overview or you get weird behavior. See https://github.com/Automattic/wp-calypso/pull/101614 and https://github.com/Automattic/wp-calypso/pull/101156 for more details.

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

This helps streamline the migration flow and gives the user fewer steps to go through.

## Testing Instructions

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

You'll need to use the calypso.live url. Testing the "authorize application" process won't work running Calypso locally.

**Using authorize application**
* Go to `/start`.
* Go through the steps and select "Import or migrate" on the goals step.
* Make sure you select a DIFM (assisted migration).
* When you get to the "Get ready for blazing fast speeds", click "Authorize access".
* When redirected, allow access to the app from your source site.
* After authorizing, you should be redirected to the site overview and see the screen shown below.

**Using share credentials**
* Go to `/start`.
* Go through the steps and select "Import or migrate" on the goals step.
* Make sure you select a DIFM (assisted migration).
* When you get to the "Get ready for blazing fast speeds", click "Share credentials instead".
* Enter your source site credentials.
* After sharing, you should be redirected to the site overview and see the screen shown below.

![CleanShot 2025-03-26 at 11 36 40](https://github.com/user-attachments/assets/3f9cf758-8a4f-4109-956c-871679b745fa)


## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

markbiek committed:
Redirect from the site credentials step directly to /sites/overview

This is the first step towards removing the SITE_MIGRATION_ASSISTED_MIGRATION step.

markbiek committed:
Redirect from the site authorization step directly to /sites/overview

This is the first step towards removing the SITE_MIGRATION_ASSISTED_MIGRATION step.

markbiek committed:
Fixing site-migration tests after removing assisted migration step

markbiek committed:
Remove references to ASSISTED_MIGRATION step

We'll delete the actual files in a separate PR

markbiek committed:
Copy updates on the migration-started-difm

andres-blanco commented on client/landing/stepper/declarative-flow/flows/site-migration-flow/site-migration-flow.ts:
```diff
@@ -498,37 +503,33 @@ const siteMigration: Flow = {
+									siteId,
+									from: from || fromQueryParam,
+									siteSlug,
+									preventTicketCreation: true,
```
why is this parameter being passed to the the overview?

andres-blanco commented on client/landing/stepper/declarative-flow/flows/site-migration-flow/site-migration-flow.ts:
```diff
@@ -498,37 +503,33 @@ const siteMigration: Flow = {
+								siteId,
+								from: from || fromQueryParam,
+								siteSlug,
+								preventTicketCreation: true,
```
same as above

andres-blanco commented on client/landing/stepper/declarative-flow/flows/site-migration-flow/site-migration-flow.ts:
```diff
@@ -483,10 +482,16 @@ const siteMigration: Flow = {
+								siteId,
+								from: from || fromQueryParam,
+								siteSlug,
+								preventTicketCreation: true,
```
should this be an argument to the overview?

andres-blanco commented on client/landing/stepper/declarative-flow/flows/site-migration-flow/site-migration-flow.ts:
```diff
@@ -586,10 +587,16 @@ const siteMigration: Flow = {
+								siteId,
+								from: from || fromQueryParam,
+								siteSlug,
+								preventTicketCreation: true,
```
same as above

markbiek commented:
@andres-blanco I wasn't sure about those parameters. We were passing them to the step that we're now skipping so I included them when redirecting to the overview.

But it may be that we only need the `ref` and not the others.

andres-blanco commented:
This is working in general. I think that `preventTicketCreation` should not be passed as a query param to `/overview`. That query param was related to the internal working of the message step and is not relevant to the overview.
It might make sense to analyze what is actually needed by `/overview` and only add that

markbiek commented:
> This is working in general. I think that `preventTicketCreation` should not be passed as a query param to `/overview`. That query param was related to the internal working of the message step and is not relevant to the overview. It might make sense to analyze what is actually needed by `/overview` and only add that

Got it, thanks for the context. I'll get that cleaned up!

markbiek committed:
Remove unneeded query args when redirecting to the overview

markbiek commented:
@andres-blanco I verified that the only thing we want to include is the `ref` parameter. I've pushed updates to remove the other args.

daledupreez commented on client/landing/stepper/declarative-flow/flows/site-migration-flow/site-migration-flow.ts:
```diff
@@ -57,7 +57,6 @@ const siteMigration: Flow = {
 			STEPS.SITE_MIGRATION_ASSIGN_TRIAL_PLAN,
 			STEPS.SITE_MIGRATION_INSTRUCTIONS,
 			STEPS.ERROR,
-			STEPS.SITE_MIGRATION_ASSISTED_MIGRATION,
```
Question: can we remove the step definition entirely in a follow-up PR? It's likely this is the only code using that step.

markbiek replied:
Yes! That was exactly my plan! 😀

markbiek replied:
I've got a PR ready once this PR is merged.

https://github.com/Automattic/wp-calypso/pull/101955

daledupreez commented on client/landing/stepper/declarative-flow/flows/site-migration-flow/site-migration-flow.ts:
```diff
@@ -483,50 +477,19 @@ const siteMigration: Flow = {
-								{ siteId, siteSlug, from: fromQueryParam, error: hasError },
-								STEPS.SITE_MIGRATION_CREDENTIALS.slug
-							)
-						);
+						return exitFlow( addQueryArgs( { ref: 'site-migration' }, `/overview/${ siteSlug }` ) );
```
This `if` block looks redundant - I think we can remove the entire block and just call `exitFlow()` below.

daledupreez commented on client/landing/stepper/declarative-flow/flows/site-migration-flow/site-migration-flow.ts:
```diff
@@ -558,7 +521,7 @@ const siteMigration: Flow = {
-					const { action, authorizationUrl, from } = providedDependencies as {
+					const { action, authorizationUrl } = providedDependencies as {
 						action: string;
 						from: string;
```
I assume we can drop the `from` from the type here.

daledupreez commented on client/landing/stepper/declarative-flow/flows/site-migration-flow/site-migration-flow.ts:
```diff
@@ -586,12 +549,7 @@ const siteMigration: Flow = {
 					}
 
 					//TODO: Add a skip flag to track the user is having trouble to share the credentials.
```
Can we remove this TODO?

daledupreez commented on client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-application-password-authorization/index.tsx:
```diff
@@ -55,9 +60,10 @@ const SiteMigrationApplicationPasswordsAuthorization: Step< {
 
 	useEffect( () => {
 		if ( isStoreApplicationPasswordSuccess ) {
+			dispatch( resetSite( siteId ) );
```
Can we make the call conditional on the value of `siteId`? Or at least require the value to be non-zero.
```suggestion
			siteId && dispatch( resetSite( siteId ) );
```

daledupreez commented on client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-credentials/index.tsx:
```diff
@@ -64,6 +67,7 @@ const SiteMigrationCredentials: Step< {
 		applicationPasswordsInfo?: ApplicationPasswordsInfo
 	) => {
 		const action = getAction( siteInfo, applicationPasswordsInfo );
+		dispatch( resetSite( siteId ) );
```
As above, I think we need to check that we have a siteId first.
```suggestion
		siteId && dispatch( resetSite( siteId ) );
```

daledupreez reviewed (commented):
I won't test quite yet, but something I am not clear on is around when and where we expect ticket creation to happen for HE-assisted migrations. It doesn't look like that has been touched in this PR, so it's less obvious what the expectations are with the removed step.

markbiek committed:
Remove unneeded if block

markbiek committed:
Remove old TODO

markbiek committed:
Remove unused type property

markbiek committed:
Ensure we have a valid siteId before resetting the store

markbiek commented:
> I won't test quite yet, but something I am not clear on is around when and where we expect ticket creation to happen for HE-assisted migrations. It doesn't look like that has been touched in this PR, so it's less obvious what the expectations are with the removed step.

I spoke with @andres-blanco and it appears that having ticket creation happen on the importer-migrate-message step was the original behavior. Then ticket creation moved to the credentials form but was left on the importer-migrate-message step for backwards compatibility while things were still in flux. At this point, it's safe to remove the step.

p1743076824277849-slack-C0Q664T29

markbiek committed:
Update so the 'Let us guide you' link creates the ZD ticket

Previously clicking that link would send you to the importer-migrate-message step which would create the ticket. Now that step is gone so we need to handle it from the credentials step (before we redirect to /overview)

markbiek commented:
@andres-blanco I think this is ready for another review when you have time.

andres-blanco commented on client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-credentials/index.tsx:
```diff
@@ -51,18 +57,47 @@ const SiteMigrationCredentials: Step< {
+	const siteSlugParam = useSiteSlugParam();
+	const fromUrl = useQuery().get( 'from' ) || '';
+	const siteSlug = siteSlugParam ?? '';
+	const shouldPreventTicketCreation = useQuery().get( 'preventTicketCreation' ) === 'true';
```
do we need this query param? It doesn't seem to be used. Also, this will not prevent ticket creation if the action is different from `skip`, so if needed, the it needs to be renamed.

markbiek replied:
I've renamed it.

andres-blanco commented on client/landing/stepper/declarative-flow/internals/steps-repository/site-migration-credentials/index.tsx:
```diff
@@ -72,14 +107,27 @@ const SiteMigrationCredentials: Step< {
 	useEffect( () => {
 		if ( siteId ) {
-			updateMigrationStatus( { status: MigrationStatus.PENDING_DIFM } );
+			updateMigrationStatus( { status: MigrationStatus.STARTED_DIFM } );
```
why this change?

markbiek replied:
It used to be that we set `PENDING_DIFM` in this step and then changed to `STARTED_DIFM` in the `importer-migrate-message` step.

Now that we're skipping `importer-migrate-message`, I thought it made sense to set the status here since of in `/overview.`

markbiek committed:
Rename ticket creation skip variable

markbiek committed:
Fix translation warning on new subtitle copy

markbiek committed:
Remove ticket creation prevention from 'Let us guide you' link

andres-blanco reviewed (approved):
Ok, I tested this four paths:
* Application passwords
* Credentials
* Skipping
* Selecting credentials and then clicking the I need help link

In all of these cases I got redirected to the correct /overview page and the ZD ticket was created.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/17319760

Some locales (Hebrew, Japanese) have been temporarily machine-translated due to translator availability. All other translations are usually ready within a few days. Untranslated and machine-translated strings will be sent for translation next Monday and are expected to be completed by the following Friday.

Thank you @markbiek for including a screenshot in the description! This is really helpful for our translators.
